packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -363,30 +362,41 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
 
     function getRequiredCollateral(
         uint256 _principalAmount,
-        uint256 _maxPrincipalPerCollateralAmount,
+        uint256 _maxPrincipalPerCollateralAmount, //should be expressed expanded by collateralTokenDecimals and principalTokenDecimals 
         CommitmentCollateralType _collateralTokenType,
-        address _collateralTokenAddress
+        address _collateralTokenAddress,
+        address _principalTokenAddress
     ) public view virtual returns (uint256) {
         if (_collateralTokenType == CommitmentCollateralType.NONE) {
             return 0;
         }
+
+        uint8 collateralDecimals = 0;
+        uint8 principalDecimals = 0;
+
```

Comment:
```suggestion
        uint8 principalDecimals = IERC20MetadataUpgradeable(_principalTokenAddress)
            .decimals();
        uint8 collateralDecimals = 0;
```

Reply:
need to use an if statement around the IERC20 call here still . Dont think we need to change anything.